compiler/ilgen/IlBuilder.cpp

```diff
@@ -1494,9 +1494,8 @@ IlBuilder::integerizeAddresses(TR::IlValue **leftPtr, TR::IlValue **rightPtr)
    {
    TR::IlValue *left = *leftPtr;
    TR::IlValue *right = *rightPtr;
-   if (left->getSymbol()->getDataType() == TR::Address)
+   if (left->getSymbol()->getDataType() == TR::Address && right->getSymbol()->getDataType() == TR::Address)
```

Comment:
you're right about that though the code was in a weirder state to start with. Technically, this integerizeAddresses joker was created for the GT,GE,LT,LE comparisons against addresses, because the OMR compiler doesn't actually define all the opcodes needed to compare against opcodes. I'll take another look at this aspect of the change to see if I can improve it.

Thanks for keeping me honest, @Leonardo2718 !